src/PhpParser/Comparing/ConditionSearcher.php

```diff
@@ -67,6 +79,15 @@ private function searchElseForVariableRedeclaration(Assign $assign, Else_ $else)
                 continue;
             }
 
+            $isFoundInExpr = (bool) $this->betterNodeFinder->findFirst(
+                $statementElse->expr->expr,
```

Comment:
What if `$isFoundInExpr` is true for `$statementIf` but not for `$statementElse`?

Reply:
that's need another failing fixture, please create new PR for that, feel free cherry-pick my commits as well.